### PR TITLE
Rebase for serenity:0.11.4 and songbird:0.3.0

### LINKED ISSUE #146: Allow disabling the DJ role
## Rationale

When the DJ roles were introduced, no alternative for keeping the old behavior was implemented (besides manually assigning the role to all members of a guild).

## Description

Allow, through a configuration command, disabling the DJ role as a requirement to use restricted commands. This allows a guild to keep the old bot behavior, should they wish so. Additionally, if implemented, make it the default behavior (so that the role isn't created unnecessarily).

## Reference

[feat: custom DJ Role for parrot commands #115](https://github.com/aquelemiguel/parrot/pull/115)

### LINKED ISSUE #196: Command permission update procedure has changed
### 📝 Description

Looks like discord updated the way command permissions are handled:
https://discord.com/developers/docs/change-log

Now it requires the use of a bearer token:
https://discord.com/developers/docs/topics/oauth2#client-credentials-grant

Serenity seems to have some level of support for this:
https://github.com/serenity-rs/serenity/blob/f82a16dae0c5b1910598115b264a66fb8182a340/src/model/application/oauth.rs#L11-L12

Not entirely sure about implementation though.

The bot seems to choke on these changes:
```
thread 'tokio-runtime-worker' panicked at 'failed to create command permission: Http(UnsuccessfulRequest(ErrorResponse { status_code: 403, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("discord.com")), port: None, path: "/api/v9/applications/<>/guilds/<>/commands/<>/permissions", query: None, fragment: None }, error: DiscordJsonError { code: 20001, message: "Bots cannot use this endpoint", errors: [] } }))', src/handlers/serenity.rs:107:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'tokio-runtime-worker' panicked at 'failed to create response: Serenity(Http(UnsuccessfulRequest(ErrorResponse { status_code: 404, url: Url { scheme: "https", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("discord.com")), port: None, path: "/api/v9/interactions/<>/<>/callback", query: None, fragment: None }, error: DiscordJsonError { code: 10062, message: "Unknown interaction", errors: [] } })))', src/handlers/serenity.rs:462:18
```

Any command that requires permissions to use is currently unavailable.

### 🪜 Reproduction Steps

1. Spin up the docker images with required environment variables
2. Attempt to use any command that requires/sets it's own permissions

### ℹ Environment / Computer Info

- **Parrot version**: ` ghcr.io/aquelemiguel/parrot:latest`
- **Operating System**: ` ghcr.io/aquelemiguel/parrot:latest`

### 📸 Screenshots

_No response_

### PR DESCRIPTION
**Pull request by StaticRocket**
Well, we should be good to go. There's a lot here. Needs some reviews before I'm comfortable with it.

This just rips out the permission system entirely right now. We should set some safe defaults later.

Resolves #196 
Resolves #146

### CODE REVIEW

**Comment by StaticRocket** [comment]
Getting closer. Only obvious thing left is some weird behavior when the user isn't in the same voice channel as the bot. May have mangled permissions somewhere... I'll sort it tomorrow.

**Comment by StaticRocket** [comment]
@afonsojramos a review when you have time would be great!

**Review by joao-conde — APPROVED** [review]

**Comment by joao-conde** [comment]
@StaticRocket alter the workflows too to bump minimal rust version

**Comment by StaticRocket** [comment]
Huh, I didn't know that information is stored repos now.

**Comment by StaticRocket** [comment]
What is the best way to add cmake to the base image the docker workflow uses?

**Comment by afonsojramos** [comment]
> What is the best way to add cmake to the base image the docker workflow uses?

I can take a look in a bit if that's the true issue. But if it is just that, it should be easy!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@ jobs:
     name: Build
     strategy:
       matrix:
-        rust-version: ["1.58", "stable"]
```
Comment (joao-conde) [inline .github/workflows/main.yml:10]: The idea is to run for the minimal Parrot supports and for the current stable one.
Have you tried 1.59? If it works, change this to 1.59, if not 1.60, if not 1.61 and if not 1.62

**Review by joao-conde — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 1 above)
Comment (StaticRocket) [reply .github/workflows/main.yml:10]: Ah, alright. I thought we were just trying to keep up with latest stable release.

Comment (joao-conde) [reply .github/workflows/main.yml:10]: That's the "stable" portion yeah

**Comment by StaticRocket** [comment]
I'm an idiot. I looked at the docker file the other day to try and change the rust version and completely ignored the apt commands specifically installing necessary build tools. We should be good now, cmake is here.

**Comment by afonsojramos** [comment]
It seems that it was [audiopus_sys@v0.2.0](https://github.com/Lakelezz/audiopus_sys/releases/tag/v0.2.0) that started having the `cmake` dependency.

**Review by afonsojramos — APPROVED** [review]
Looks good to me!

**Review by joao-conde — APPROVED** [review]

**Pull request merged**